packages/api/src/webhooks/cannon.ts

```diff
@@ -290,43 +375,102 @@ export default class WebhookCannon {
       // sign payload if there is a webhook secret
       if (webhook.sharedSecret) {
         let signature = sign(params.body, webhook.sharedSecret);
-        params.headers["Livepeer-Signature"] = `t=${timestamp},v1=${signature}`;
+        params.headers[SIGNATURE_HEADER] = `t=${timestamp},v1=${signature}`;
+      }
+      try {
+        await this.storeTriggerTime(webhook);
+      } catch (e) {
+        console.log(
+          `Unable to store trigger time of webhook ${webhook.id} url: ${webhook.url}`
+        );
       }
 
+      let resp: Response;
+      let errorMessage;
+      let statusCode;
+
       try {
         logger.info(`webhook ${webhook.id} firing`);
         const startTime = process.hrtime();
-        let resp = await fetchWithTimeout(webhook.url, params);
-        await this.storeResponse(webhook, event, resp, startTime);
+        resp = await fetchWithTimeout(webhook.url, params);
+        const responseBody = await resp.text();
+        await this.storeResponse(webhook, event, resp, startTime, responseBody);
+        statusCode = resp.status;
+
         if (resp.status >= 200 && resp.status < 300) {
           // 2xx requests are cool. all is good
           logger.info(`webhook ${webhook.id} fired successfully`);
           return true;
         }
-
+        errorMessage = responseBody;
         if (resp.status >= 500) {
-          await this.retry(trigger);
+          await this.retry(
+            trigger,
+            params,
+            new Error("Status code: " + resp.status)
+          );
         }
 
         console.error(
           `webhook ${webhook.id} didn't get 200 back! response status: ${resp.status}`
         );
         // we don't retry on non 400 responses. only on timeouts
         // this.retry(event);
-        return;
       } catch (e) {
         console.log("firing error", e);
-        await this.retry(trigger);
+        errorMessage = e.message;
+        await this.retry(trigger, params, e);
+      } finally {
+        if (statusCode >= 300 || !statusCode) {
```

Comment:
Hmm, since we execute this in a `finally` anyway, which will always get executed, I think it might be cleaner to update both the `lastTriggerTime` and the `lastFailure` on a single go, WDYT? Then we can have a single function that receives the timestamp (either from before making the request or after), and whether there was any errors and its status code etc, then we always update the status code and maybe update the `lastFailure` depending on there being an error or not. Doesn't seem like it would be a big refactor here in the end.